src/components/UI/Card.js

```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import { VscEdit } from "react-icons/vsc";
+import { VscSave } from "react-icons/vsc";
+import { VscClose } from "react-icons/vsc";
+
+import "./Card.css";
+
+class Card extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      checked: false,
+      clicked: false,
+      inputValue: "",
+      textareaValue: "",
+      input: "",
+      textarea: ""
+    };
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    this.setState({input: this.state.inputValue});
+    this.setState({textarea: this.state.textareaValue});
+  }
+
+  inputChangeHandler(event) {
+    this.setState({ inputValue: event.target.value });
+  }
+
+  textareaChangeHandler(event) {
+    this.setState({ textareaValue: event.target.value });
+  }
+
+  render() {
+    return (
+      <div className={this.state.checked ? "card-item" : "card"}>
+        <form onSubmit={(e) => this.handleSubmit(e)}>
+          <div className={"card-inner__caption"}>
+            <input
+              autoComplete="off"
+              className={
+                this.state.clicked
+                  ? "card-text__caption"
+                  : "card-text__caption-none"
+              }
```

Comment:
Here you should better use extra package:
https://www.npmjs.com/package/classnames
It will allow to add more flexibility to conditional styling.